pkg/api/types.go

```diff
@@ -385,6 +388,9 @@ const (
 	// BetaStorageClassAnnotation represents the beta/previous StorageClass annotation.
 	// It's currently still used and will be held for backwards compatibility
 	BetaStorageClassAnnotation = "volume.beta.kubernetes.io/storage-class"
+	// AlphaStorageTopologyAnnotation is set in the StorageClass and gives the scheduler
+	// a hint that the volumes in this StorageClass have a topology constraint
+	AlphaStorageTopologyAnnotation = "storage.alpha.kubernetes.io/topology-key"
```

Comment:
As discussed offline, I think we can use labels for expressing topology. Here is the algorithm that we discussed:

1. PVs expose a single topology label - `topology.kubernetes.io/node:foo`, or `topology.k8s.io/zone:bar`
2. PVCs can already select using this topology label. Storage Classes should include a Label Selector that can specify topology constaints.
```yaml
kind: StorageClass
spec:
  selector:
    - key: "topology.k8s.io/node"
```
or 
```yaml
kind: StorageClass
spec:
  selector:
    - key: "topology.k8s.io/zone"
      operator: In
      values:
      - bar-zone
```
3. Nodes will expose all aspects ot topology via consistent label keys.
```yaml
kind: Node
metadata:
  label: 
   topology.kubernetes.io/node : foo
   topology.k8s.io/zone : bar
```
4. The scheduler can then combine NodeSelector on the pod with the Selector on the StorageClass while identifying Nodes that can meet the storage locality requirements. 

This method would require using consistent label keys across nodes and PVs. I hope that's not a non starter. 

cc @thockin Does this align with your thoughts on topology?